src/Fhir.Anonymizer.Core/PartitionedExecution/FhirPartitionedExecutor.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Fhir.Anonymizer.Core.PartitionedExecution;
+
+namespace Fhir.Anonymizer.Core
+{
+    public class FhirPartitionedExecutor
+    {
+        public IFhirDataReader RawDataReader { set; get; }
+
+        public IFhirDataConsumer AnonymizedDataConsumer { set; get; }
+
+        public Func<string, string> AnonymizerFunction { set; get; }
+
+        public int PartitionCount { set; get; } = Constants.DefaultPartitionedExecutionCount;
+
+        public int BatchSize { set; get; } = Constants.DefaultPartitionedExecutionBatchSize;
+
+        public FhirPartitionedExecutor(IFhirDataReader rawDataReader, IFhirDataConsumer anonymizedDataConsumer, Func<string, string> anonymizerFunction)
+        {
+            RawDataReader = rawDataReader;
+            AnonymizedDataConsumer = anonymizedDataConsumer;
+            AnonymizerFunction = anonymizerFunction;
+        }
+
+        public async Task ExecuteAsync(CancellationToken cancellationToken, bool breakOnAnonymizationException = false, IProgress<BatchAnonymizeResult> progress = null)
+        {
+            Queue<Task<IEnumerable<string>>> executionTasks = new Queue<Task<IEnumerable<string>>>();
+            List<string> batchData = new List<string>();
+
+            string content;
+            while ((content = await RawDataReader.NextAsync().ConfigureAwait(false)) != null)
```

Comment:
Can we merge into a line?

Reply:
I'm afraid we cannot, it not support new object in while condition.